src/other/codemirror/CodeMirror.tsx

```diff
@@ -35,31 +34,71 @@ export interface CodeMirrorProps {
 
 export interface CodeMirrorState {
     isLoaded: boolean;
+    content: string;
     isEditing: boolean;
 }
 
 class CodeMirror extends React.Component<CodeMirrorProps, CodeMirrorState> {
     constructor(props: CodeMirrorProps) {
         super(props);
-        this.state = { isLoaded: false, isEditing: false };
+        this.state = {
+            content: "",
+            isLoaded: false,
+            isEditing: false,
+        };
         this.handleContentChange = this.handleContentChange.bind(this);
+        this.handleFocus = this.handleFocus.bind(this);
+        this.handleBlur = this.handleBlur.bind(this);
+    }
+
+    componentDidMount() {
+        const { content } = this.props;
+        this.setState({ content: content || "" });
+    }
+
+    componentDidUpdate(prevProps: CodeMirrorProps) {
+        const { content } = this.props;
+        if (prevProps.content !== content) {
+            this.setState({ content: content || "" });
+        }
     }
 
     /*
      * editor - CodeMirror object https://uiwjs.github.io/react-codemirror/
      * viewUpdate - object containing the update to the editor tree structure
      */
-    handleContentChange(newContent: string, viewUpdate: { origin: string }) {
```

Comment:
This `viewUpdate: { origin: string }` doesn't exist in CodeMirror 6, and WebApp and tests work fine with this parameter removed